### PR TITLE
Add eslint and fix eslint issues

### LINKED ISSUE #46: Add eslint
Some suggestion for pure javascript

Hook up eslint and prettier and unicorn. [Unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) has sane defaults for modern javascript.
See my PR here https://github.com/updatecli/updatecli-action/pull/100

Add `jsconfig.json` should help with improve type detection for vscode.
Add `jsconfig.json` in your test folder that helps with jest type annotations.

`index.js` should really only be routing probot.

Everything else should be imported. FYI probot blocks you from switching to ESM. So stick with commonjs due to: https://github.com/probot/pino/pull/188
Because of it you need to turn off some features of unicorn, see [release-drafter eslintrc](https://github.com/release-drafter/release-drafter/blob/f4824e6ad8f4ac1e687db104bc43dfef0febaba7/.eslintrc#L13)

_Originally posted by @jetersen in https://github.com/timja/github-comment-ops/issues/7#issuecomment-1192299304_

### PR DESCRIPTION
**Pull request by jetersen**
fixes #46 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by timja** [comment]
lint issues

File: package.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
   "scripts": {
     "start": "node index.js",
     "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js",
-    "lint": "prettier --check .",
-    "lint:fix": "prettier --write ."
+    "lint": "eslint .",
+    "lint:fix": "eslint --fix ."
```
Comment (timja) [inline package.json:10]: does this make it so you don't need to invoke prettier at all? i.e. the plugin will do prettier under the hood?

File: app/converters.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 export function extractUsersAndTeams(orgName, reviewers) {
-  const split = reviewers.split(/[,\s]/);
+  const split = reviewers.split(/[\s,]/).filter(Boolean);
```
Comment (timja) [inline app/converters.js:2]: what does this `.filter(Boolean)` do?

File: package.json (hunk 1 above)
Comment (jetersen) [reply package.json:10]: Yup!

File: app/converters.js (hunk 2 above)
Comment (jetersen) [reply app/converters.js:2]: filters out empty strings and empty entries

Comment (lemeurherve) [reply app/converters.js:2]: Interesting but not obvious, maybe add a comment about it?

Comment (jetersen) [reply app/converters.js:2]: JavaScript is not obvious however this is basic type coercion in JavaScript

Comment (jetersen) [reply app/converters.js:2]: Created tests here for it: https://github.com/timja/github-comment-ops/pull/52

**Comment by timja** [comment]
Thanks!

File: app/converters.js (hunk 2 above)
Comment (timja) [reply app/converters.js:2]: thanks!

File: package.json (hunk 1 above)
Comment (jetersen) [reply package.json:10]: however this only applies to js files 😅

So potentially you want it for other files

Comment (timja) [reply package.json:10]: https://github.com/timja/github-comment-ops/issues/57

**Pull request merged**